src/vs/workbench/contrib/search/browser/searchActions.ts

```diff
@@ -549,9 +587,27 @@ export class ReplaceAction extends AbstractSearchAndReplaceAction {
 
 	override async run(): Promise<any> {
 		this.enabled = false;
+		let elementsToReplace: RenderableMatch[] = [];
+		const currentSelection: RenderableMatch[] = [];
+
+		this.viewer.getSelection().forEach((elem) => {
+			currentSelection.push(<RenderableMatch>elem);
+		});
+
+		// if you're using multiselect and you click to replace one element from the tree, replace all selected
+		if (currentSelection.length > 1 && currentSelection.includes(this.element)) {
+			elementsToReplace = currentSelection;
+		} else {
+			elementsToReplace.push(this.element);
+		}
+
+		elementsToReplace.forEach(async (elem) =>
```

Comment:
forEach isn't async, this isn't waiting on the inner promise. You can use a for loop or do them all simultaneously and wait with `Promise.all`